Source/OxyPlot.ImageSharp/PngRenderingContext.cs

```diff
@@ -119,27 +375,154 @@ public override OxySize MeasureText(string text, string fontFamily = null, doubl
+                    for (int x = 1; x < sourceImage.Width - 1; x++)
+                    {
+                        sourceImage[x, 0] = sourceImage[x, 1];
+                        sourceImage[x, sourceImage.Height - 1] = sourceImage[x, sourceImage.Height - 2];
+                    }
+                    for (int y = 1; y < sourceImage.Height - 1; y++)
+                    {
+                        sourceImage[0, y] = sourceImage[1, y];
+                        sourceImage[sourceImage.Width - 1, y] = sourceImage[sourceImage.Width - 2, y];
+                    }
+                }
+
+                sourceImage.Mutate(img =>
+                {
+                    img.Transform(rescale, resampler);
+                    destRough.Intersect(sourceImage.Bounds());
+                    img.Crop(destRough);
+                });
+
+                image.Mutate(img =>
+                {
+                    img.DrawImage(sourceImage, new Point((int)dest.X, (int)dest.Y), new GraphicsOptions(interpolate, (float)opacity));
+                });
+            }
+            catch (ImageProcessingException)
+            {
+                // Swallow: it's probably because we are trying to render outside of the image: https://github.com/SixLabors/ImageSharp/pull/877
+                // TODO: verify that we are trying to render outside of the image... somehow
+                //  - I don't think this can be done without having to track the ImageSharp code unhealthily closely
+            }
+        }
+
+        private class EdgeClampedResampler : SixLabors.ImageSharp.Processing.Processors.Transforms.IResampler
+        {
+            public float Radius => throw new NotImplementedException();
+
+            public float GetValue(float x)
+            {
+                throw new NotImplementedException();
+            }
         }
 
         /// <inheritdoc/>
         public override void DrawLine(IList<ScreenPoint> points, OxyColor stroke, double thickness, EdgeRenderingMode edgeRenderingMode, double[] dashArray, LineJoin lineJoin)
         {
             image.Mutate(img =>
             {
-                var brush = Brushes.Solid(new Rgba32(stroke.R, stroke.G, stroke.B, stroke.A));
-                PointF[] mappedPoints = points.Select(p => new PointF((float)p.X, (float)p.Y)).ToArray();
+                var actualThickness = this.GetActualThickness(thickness, edgeRenderingMode);
+                var actualDashArray = dashArray != null
+                    ? this.ConvertDashArray(dashArray, actualThickness)
+                    : null;
+
+                var pen = actualDashArray != null
+                    ? new Pen(ToRgba32(stroke), actualThickness, actualDashArray)
+                    : new Pen(ToRgba32(stroke), actualThickness);
+                var actualPoints = this.GetActualPoints(points, thickness, edgeRenderingMode).ToArray();
```

Comment:
I think you have to pass the `actualThickness` here.

Reply:
Actually, `GetSnapOffset` inside `GetActualPoints` performs the transform itself. This is consistent with `SkiaSharp`

Reply:
Oh ok sorry, my mistake then.